sympy/series/formal.py

```diff
@@ -974,6 +1020,17 @@ def infinite(self):
 
     def _get_pow_x(self, term):
         """Returns the power of x in a term."""
+        ind = self.ind
+        for t in Add.make_args(ind):
+            if isinstance(t, Mul):
+                for subt in Mul.make_args(t):
+                    if (isinstance(subt, Pow) and subt.exp.is_symbol):
+                        return S.One
+            else:
+                if isinstance(t, Pow):
+                    if isinstance(t.exp, Mul) or t.exp.is_symbol:
+                        return S.One
```

Comment:
Can the conditions in L1030 and L1031 be written as, 
```
if isinstance(t, Pow) and (isinstance(t.exp, Mul) or t.exp.is_symbol):
``` 
I believe that nesting should be avoided wherever necessary. It helps in debugging using a debugger and enhances readability. If you follow some other principles then no issues.

Reply:
Nestings have been removed. Thanks a lot @czgdp1807